common/src/main/java/io/netty/util/concurrent/DefaultThreadFactory.java

```diff
@@ -84,7 +82,7 @@ private static String toPoolName(Class<?> poolType) {
         }
     }
 
-    public DefaultThreadFactory(String poolName, boolean daemon, int priority, ThreadGroup threadGroup) {
+    public DefaultThreadFactory(String poolName, boolean daemon, int priority) {
```

Comment:
@normanmaurer I'm curious if this is truly a problem (I trust you that it is, I just want to understand :smile:)? This is a relatively recent change to `DefaultThreadFactory`, it's not used internally. Are there known external uses of this?

*If* the API can not be broken, then the *default* constructors should not lead to `DefaultThreadFactory` being sticky about the thread group. This means that the restriction on the field `threadGroup` being `null` should be removed as passing `null` down through the thread constructors will still lead to the default thread group being used which is what is really needed here for cases like `GlobalEventExecutor` and `ThreadDeathWatcher`. I *can* push another commit if needed.